src/analyses/variable-sensitivity/interval_abstract_value.h

```diff
@@ -27,15 +27,14 @@ class interval_abstract_valuet : public abstract_value_objectt
     interval_abstract_value_pointert;
 
 public:
-  explicit interval_abstract_valuet(typet t);
-  interval_abstract_valuet(typet t, bool tp, bool bttm);
+  explicit interval_abstract_valuet(const typet &t);
+  interval_abstract_valuet(const typet &t, bool tp, bool bttm);
 
-  interval_abstract_valuet(const constant_interval_exprt e, int merge_count);
-
-  explicit interval_abstract_valuet(const constant_interval_exprt e);
+  explicit interval_abstract_valuet(const constant_interval_exprt &e);
+  interval_abstract_valuet(const constant_interval_exprt &e, int merge_count);
```

Comment:
What does `merge_count` do?

Reply:
It's used to drive a repeated merged interval to TOP, but for nothing else. It's not clear to me why we would do this, though. 
```
interval_abstract_valuet::interval_abstract_valuet(
  const constant_interval_exprt &e,
  int merge_count)
  : abstract_value_objectt(
      e.type(),
      e.is_top() || merge_count > 10,
      e.is_bottom()),
    interval(e),
    merge_count(merge_count)
{
}
```

Reply:
@jezhiggins IIRC this was for loop bodies, e.g. consider

```
for(int i = 0; i < 2; ++i)
{
  // do something
}
```

We start out with `i in {0}`, then we get `i in {0,1}`, `i in {0,1,2}` etc, because we keep merging with the loop body rather than inlining the loop like we do in cbmc, so we never reach a natural termination condition. There's probably a better way to do this than this hack though.

Reply:
Right!  It is a way of implementing widening!  Ofcourse!

@hannes-steffenhagen-diffblue thanks for the input.  I believe there is a better way and it is on the tasks I've asked @jezhiggins to look at.  Briefly... `ai_historyt` and it's children have a `bool should_widen() const` method which allows CFG aware decisions on when to widen.  So we can widen on the 10th iteration or on recursive function calls.  Domains that have non-trivial height / want to do fancy widening things can then query this in their `merge` method and take appropriate action.  VSD needs to do this and then pass a "widen" parameter down to the `abstract_objectt::merge` method.

Note that this PR and other changes to `value_sett` are working on adding a size limit that is independent of widening so that the domain will be space limited not just at merge but also at transform.

TL;DR : this can go; there are better ways of doing this.

Reply:
It's gone!